### PR TITLE
Comments and thread safety

### PR DESCRIPTION
**Pull request by RogerTangos**
Hey Bart,

Your code's really cool. I actually learned learned a lot by looking at it 😃! Here are some comments:

- Setting a function named `_` is extremely non-conventional. In Javascript, that type of stuff happens all the time (i.e. the underscore library), but in Python, it's strongly frowned upon. I'd have given this an explicit name like `translate`
- Overriding the global `_` is even less conventional 😋 . If I were to do that in a repo, I'd make sure to have a bulletproof justification for it either in the PR or in the comments.
- I realized that by overriding `_` globally when a user messaged, it was possible that two different-language users messaging simultaneously could receive messages in the wrong language. However, since all the `_` function is doing is allowing gettext to select the correct language, why not modify the wrapper to inject a new attribute `lang` into the functions it's translating?  **That's the first commit.**

- It's only a small step from there to realize that the language (which is from the context) doesn't even need a wrapper, because context can be passed directly to the translate function. **that's the second commit**
- Of course, it's also possible to use the wrapper so that it injects a translate function that already has the correct language set. In my view, there's not too much point to that, unless you're just trying to learn about wrapper functions... In that case, that is left as an exercise for the reader 😆

### CODE REVIEW

**Comment by bartsopers** [comment]
Thanks a lot for the improvements and the explanation :-) It looks a lot better now. 

The only concern that I have is that we cannot automatically extract strings from the Documents enum anymore, as these aren't wrapped in the `translate` function anymore. This was the reason for overriding the global _ method in documents.py, I got it from here: https://docs.python.org/3/library/gettext.html#deferred-translations

File: documents.py [hunk 1]
```diff
@@ -2,27 +2,22 @@
...
         "filename": "xr_structures.pdf",
     }
     XR_UNOFFICIAL_GUIDE = {
-        "description": _("But really, how does this work? (the reality)"),
+        "description": "But really, how does this work? (the reality)",
         "filename": "xr_unofficial_guide.pdf",
     }
-
-
-del _
```
Comment (bartsopers) [inline documents.py:28]: Due to the removal of the `_` (or, in the new situation, `translate`) function here, we cannot automatically extract strings for translations from this file anymore (using tools like gettext). 
I got the idea for these 'deferred translations'  from here: https://docs.python.org/3/library/gettext.html#deferred-translations
Could you please have a look and see if we can restore this functionality?

**Review by bartsopers — CHANGES_REQUESTED** [review]
Thanks a lot for the improvements and the explanation! The code looks a lot better now and I learned a thing or two :-) 
I added one comment in documents.py, could you please have a look at that?

File: documents.py (hunk 1 above)
Comment (bartsopers) [reply documents.py:28]: This has been fixed in #9

**Review by bartsopers — APPROVED** [review]

**Comment by RogerTangos** [comment]
I was kind of on the fence about that, since in some cases we might have the context. Still, better to keep it simple for now. I removed the translation

**Pull request merged**